### DISCUSSION TITLE
DateTime structures of choice.

### DISCUSSION THREAD

**Discussion by bobmcwhirter** [guacsec/trustify#146] [Tenets]
Which datetime-related structures should we be using?

A recent PR by @ctron has

```
-   published_at: DateTimeUtc,
+   published_at: time::OffsetDateTime,
```

Is there a tenet here we should spread more widely?

**Comment by ctron** [comment]
chrono is dead, time 0.3 so the new thing. … time has `231,569,495` downloads, chrono has `155,970,055`. 

chrono actually depends on time 0.1, adding stuff on top. But time 0.1 having fixed a CVE, didn't get updated in chrono for quite a while. That was around 2021, where many projects moved to time 0.3. Figuring out that projects actually didn't need to "on top" stuff of chrono.

And just for timestamps, I don't think we need it either. Maybe chrono updates to time 0.3 some point. By stick to time 0.1 if we don't need it.

**Reply by bobmcwhirter** [reply]
So, we should remove direct usage of chrono?

https://github.com/trustification/trustify/blob/main/Cargo.toml#L36

**Comment by gildub** [comment]
To get a bit of comparison data:

https://github.com/chronotope/chrono/issues/1423
